### DISCUSSION TITLE
Kodi 19.4 dietpi 8.11.2 cpu temp bug

### DISCUSSION THREAD

**Discussion by Thundervarela** [MichaIng/DietPi#5968] [Q&A] [answered]
Kodi 19.4 dietpi 8.11.2 cpu temp bug , missing cpu temp disable i can check it on the commandline but not on the kodi interface like this : https://imageshack.com/i/pnnyCkohj

**Comment by MichaIng** [comment, marked as answer]
I suppose it's an RPi. This is a limitation of Kodi then, not supporting the RPi sysfs entries. Or it uses the sensors package:
```sh
apt install lm-sensors
```
It this does not help we could report this to the RPi maintainer of the Kodi package. They do HW acceleration patches and such with the RPi build, so probably might add support for CPU temps as well.

**Reply by Thundervarela** [reply]
sudo apt install sensors package not found

**Reply by Thundervarela** [reply]
That's the output of that command

**Reply by Thundervarela** [reply]
i would report it if would know who is the RPi maintainer of the Kodi package and how to reacht him ...

**Reply by MichaIng** [reply]
Ah sorry, this is the correct package name:
```sh
apt install lm-sensors
```

**Reply by Thundervarela** [reply]
well i run the command apt install lm-sensors and reboot and it doesn't fix it . Already reported to https://github.com/xbmc/xbmc/issues/22240

**Reply by MichaIng** [reply]
This is the upstream repository, not the RPi specific one. Here it would be suited best: https://forums.raspberrypi.com/viewtopic.php?t=323303

And note that upstream developers usually require a lot more information to even start recognising a report, hardware, exact build, where it comes from, OS version (DietPi is irrelevant, it's Debian Bullseye, while Kodi packages come from RPi package repository) etc. But I'm quite sure they refer to downstream package maintainers back anyway, as long as you did not verify with a fresh upstream source build.

**Comment by Thundervarela** [comment]
I think i found it

**Comment by EdWaQ** [comment]
Can someone explain how I upgrade my kodi from 18.9 to 19+

**Reply by Joulinar** [reply]
First you would need to share some more details on your system

#### Required 
- DietPi version | `cat /boot/dietpi/.version`
- Distro version | `echo $G_DISTRO_NAME $G_RASPBIAN`
- Kernel version | `uname -a`
- SBC model | `echo $G_HW_MODEL_NAME` or (EG: RPi3)
- Power supply used | (EG: 5V 1A RAVpower)
- SDcard used | (EG: SanDisk ultra)

**Reply by EdWaQ** [reply]
```
root@DietPi:~# cat /boot/dietpi/.version
G_DIETPI_VERSION_CORE=8
G_DIETPI_VERSION_SUB-11
G_DIETPI_VERSION_RC=2
G_GITBRANCH='master'
G_GITOWNER='MichaIng'
G_LIVE_PATCH_STATUS[0]='not applicable'
root@DietPi:~# echo $G_DISTRO_NAME $G_RASPBIAN
buster 1
root@DietPi:~# uname -a
Linux DietPi 5.10. 103-v7l+ #1529 SMP Tue Mar 8 12:24:00 GMT 2022 armv7l GNU/Linux
root@DietPi:~# echo $G_HW_MODEL_NAME
RPi 4 Model B (armv71)
````

**Reply by MichaIng** [reply]
You need to upgrade your system to Bullseye for this: https://dietpi.com/blog/?p=811

**Reply by EdWaQ** [reply]
Thx, I'll try it soon. First of all i need a extra sd card to keep the old system

**Reply by MichaIng** [reply]
That is always a good idea 👍.

**Reply by EdWaQ** [reply]
Upgrade don't work, any error from buster to bullseye. I set a new clean dietpi on das card well done work fine.

**Reply by MichaIng** [reply]
> Upgrade don't work, any error from buster to bullseye.

Seems to late now, but if you told us the error message, we surely would have been able to solve it 😉. But a clean install is anyway recommended, just possibly more work to setup everything again.

**Reply by EdWaQ** [reply]
`libc6-dev : Broken: libgcc-8-dev (< 8.4.0-2~) but 8.3.0-6+rpi1 should be installed` this was the error after I use this code

```
# ProFTPD
[[ -f '/etc/proftpd/proftpd.conf' ]] && ! grep -q 'IfModule mod_ident.c' /etc/proftpd/proftpd.conf && sed -i '/IdentLookups/c\<IfModule mod_ident.c>\nIdentLookups off\n</IfModule>' /etc/proftpd/proftpd.conf

# Legacy network interface names
# - RPi
(( $G_HW_MODEL < 10 )) && ! grep -q 'net.ifnames=0' /boot/cmdline.txt && sed -i '/root=/s/$/ net.ifnames=0/' /boot/cmdline.txt
# - Odroid XU4
(( $G_HW_MODEL == 11 )) && ! grep -q 'net.ifnames=0' /boot/boot.ini && sed -i '/bootrootfs/s/"$/ net.ifnames=0"/' /boot/boot.ini
# - Armbian
[[ -f '/etc/armbianEnv.txt' ]] && ! grep -q 'net.ifnames=0' /etc/armbianEnv.txt && G_CONFIG_INJECT 'extraargs=' 'extraargs="net.ifnames=0"' /boot/armbianEnv.txt
```

**Reply by MichaIng** [reply]
This error is not related to the executed code but to the the `apt full-upgrade` only. The `libgcc-8-dev` package is a Buster package, and a development package anyway (not needed to "run" software, just to "build" software). So you can safely purge it:
```sh
apt autopurge libgcc-8-dev
```
